test/message/test_runner_output.out

```diff
@@ -0,0 +1,327 @@
+TAP version 13
+ok 1 - sync pass todo # TODO
```

Comment:
Looking at the [specification](https://testanything.org/tap-version-13-specification.html), shouldn't the line start with `# TODO`?

> If the directive **starts** with # TODO, the test is counted as a todo test, and the text after TODO is the explanation.

Reply:
I don't think so. If you look at the example in the spec immediately following the text you quoted:

```
not ok 13 # TODO bend space and time
```

Reply:
> I don't think so.

Indeed, the [TAP 14 specification clearly allows this use](https://github.com/TestAnything/Specification/pull/25/files#diff-1fdb6f17d18295d39b8a3ee5b260374867993b9306f3130b7df76627c60615c7R339-R341).

> having a failing TODO count as a failure, but not a TODO seems wrong.

Yes, even more so: In the examples of TAP 14 it even has [an example](https://github.com/TestAnything/Specification/pull/25/files#diff-1fdb6f17d18295d39b8a3ee5b260374867993b9306f3130b7df76627c60615c7R456-R457) noting that any TODO case will be shown as "not ok" but not treated as error.

> not ok 14 # TODO bend space and time

Reply:
It's up to the harness whether to report whether TODO tests are passing or failing, but it must not be considered a test failure to have `not ok` tests marked as SKIP or TODO.  Eg:

```$ npx tap-parser <<TAP
> TAP version 13
> not ok 1 - a todo test # TODO
> 1..1
> TAP
[
  [ 'version', 13 ],
  [
    'assert',
    Result {
      ok: false,
      id: 1,
      todo: true,
      name: 'a todo test',
      fullname: ''
    }
  ],
  [ 'plan', { start: 1, end: 1 } ],
  [ 'comment', '# todo: 1\n' ],
  [
    'complete',
    FinalResults {
      ok: true,
      count: 1,
      pass: 0,
      fail: 1,
      bailout: false,
      todo: 1,
      skip: 0,
      plan: FinalPlan {
        start: 1,
        end: 1,
        skipAll: false,
        skipReason: '',
        comment: ''
      },
      failures: [],
      time: null
    }
  ]
]```